File: demos/lua/dsa/linear/dynamic_array.lua
print("=== Lua Table as Dynamic Array ===")
local arr = {10, 20}
table.insert(arr, 30)

assert(#arr == 3, "Length assertion failed")
assert(arr[2] == 20, "Index assertion failed")
local popped = table.remove(arr)
assert(popped == 30, "Remove assertion failed")

print(string.format("Lua Table len=%d, elements=[%d, %d]", #arr, arr[1], arr[2]))
print("Lua Dynamic Array tests passed successfully.")

File: demos/python/dsa/linear/dynamic_array.py
from collections import deque

def main():
    print("=== Python list & collections.deque ===")
    arr: list[int] = [10, 20]
    arr.append(30)
    assert len(arr) == 3
    assert arr[1] == 20
    assert arr.pop() == 30

    dq: deque[str] = deque(["middle"])
    dq.appendleft("front")
    dq.append("back")
    assert list(dq) == ["front", "middle", "back"]

    print(f"Python list: {arr}, deque: {list(dq)}")
    print("Python Dynamic Array tests passed successfully.")

if __name__ == "__main__":
    main()
